### PR TITLE
Extend device.json format and initial MachXO support

### PR DESCRIPTION
**Pull request by mmicko**
- This one adds more flexible format to support MachXO, MachXO2,... models
- MachXO use new commands in bitstream
- Also MachXO needs row_bias, so new parameter is added
- cfg text file format is extended with .variant, so we can specify different sub variants per .device (which is representing chip die in this case)
- ECP5 path is not affected

### CODE REVIEW

File: libtrellis/src/Bitstream.cpp [hunk 1]
```diff
@@ -732,6 +798,65 @@ Bitstream Bitstream::serialise_chip(const Chip &chip, const map<string, string>
     // Padding
     wr.insert_dummy(ops.dummy_bytes_after_preamble);
 
+    if (chip.info.family == "MachXO") {
```
Comment (gatecat) [inline libtrellis/src/Bitstream.cpp:801]: might be neater to split this off into a `serialise_chip_machxo` function?

Comment (mmicko) [reply libtrellis/src/Bitstream.cpp]: done

**Comment by gatecat** [comment]
do you want the db pr merged first so you can then update the submodule in this?

**Comment by gatecat** [comment]
DB side has been added, so you can update this PR now.

**Comment by mmicko** [comment]
done. thanks

**Pull request merged**